subworkflows/nf-core/variantcalling/strelka/single/main.nf

```diff
@@ -2,33 +2,30 @@ include { TABIX_BGZIP as BGZIP_VC_STRELKA        } from '../../../../../modules/
 include { TABIX_BGZIP as BGZIP_VC_STRELKA_GENOME } from '../../../../../modules/nf-core/modules/tabix/bgzip/main'
 include { CONCAT_VCF as CONCAT_STRELKA           } from '../../../../../modules/local/concat_vcf/main'
 include { CONCAT_VCF as CONCAT_STRELKA_GENOME    } from '../../../../../modules/local/concat_vcf/main'
-include { STRELKA_GERMLINE                       } from '../../../../../modules/nf-core/modules/strelka/germline/main'
+include { STRELKA_GERMLINE as STRELKA_SINGLE     } from '../../../../../modules/nf-core/modules/strelka/germline/main'
 
-// TODO: Research if splitting by intervals is ok, we pretend for now it is fine.
-// Seems to be the consensus on upstream modules implementation too
 workflow RUN_STRELKA_SINGLE {
     take:
     cram                     // channel: [mandatory] [meta, cram, crai, interval.bed.gz, interval.bed.gz.tbi]
     fasta                    // channel: [mandatory]
     fasta_fai                // channel: [mandatory]
     intervals_bed_gz         // channel: [optional]  Contains a bed.gz file of all intervals combined provided with the cram input(s). Mandatory if interval files are used.
-    num_intervals            //     val: [optional]  Number of used intervals, mandatory when intervals are provided.
 
     main:
 
     ch_versions = Channel.empty()
 
-    STRELKA_GERMLINE(cram, fasta, fasta_fai)
+    STRELKA_SINGLE(cram, fasta, fasta_fai)
 
     // Figure out if using intervals or no_intervals
-    STRELKA_GERMLINE.out.vcf.branch{
-            intervals:    num_intervals > 1
-            no_intervals: num_intervals == 1
+    STRELKA_SINGLE.out.vcf.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
```

Comment:
`num_intervals` is `0` for `params.no_intervals`?
Shouldn't it be `1` as for just one big interval?

Reply:
quite a philosophical question. I was thinking of num_intervals as num_interval_files. 

This does avoid the whole weird things about querying for interval file name for the VC workflows:
```// If either no scatter/gather is done, i.e. no interval (0) or one interval (1), then don't rename samples
            new_meta.id = num_intervals <= 1 ? meta.sample : meta.sample + "_" + intervals.baseName
            new_meta.num_intervals = num_intervals

            //If no interval file provided (0) then add empty list
            intervals_new = num_intervals == 0 ? [] : intervals
            ```